Cleanup portico why-zulip markdown pages

alexandraciobica opened the pull request:
Fixes part of: https://github.com/zulip/zulip/issues/12566
* added the `.markdown` class to the pages from `why-zulip` cluster
* removed the border and box-shadow from the `img` that were added by the `.markdown` class
* removed duplicated `why-page` selectors

* added the `.markdown` class to the `integrations` page so I can deduplicate this kind of selectors: 

```
.markdown ol > li::before,
.portico-landing.integrations ol > li::before { 
```
* removed duplicated `integrations` selectors and fixed styling that was broken when I added the `markdown` class

* moved the markdown CSS to a separate file
* refactored `markdown.scss` to use SCSS nesting
* removed selectors from `why-page` that were overwritten by markdown styling
* changed the `font-size` of the markdown text 
  * here I think it looks good with `1.1rem`; it is a little bigger for `/help` and a little smaller for `why Zulip` than the current version.

**TO DO**
* move CSS of `why-page`s in a separate file (`why-pages.scss`?)
* refactor `why-pages.scss` to use SCSS nesting
* refactor `integrations.scss` to use SCSS nesting ?

**Note:**
I still have some work to do here and some testing.

timabbott commented:
Awesome, thanks for working on this @alexandraciobica!  I like a lot of the choices made here.  @rishig can you take a look at whether this has any visual regressions in comparison to the old system?  I think there's lots of follow-ups we will want to do but I think it makes sense to merge this as soon as we're happy with it.

A few notes:
* I think we should migrate that low-width `height` CSS item out of being a property of `markdown` and instead make it a property of an additional class we reference in `documentation_main.html`, e.g. `help_center_content`, since it seems pretty clearly a design element for just that page.
* Can you either post quick screenshots or do a list of what views you verified at both webapp and mobile size for /integrations, /why-zulip, /api, and /help pages just to confirm we don't have regressions there?  

I think this batch of things we should save for a follow-up PR, just to avoid having merge conflicts with any bug fixes to this PR.
* move CSS of why-pages in a separate file (why-pages.scss?)
* refactor why-pages.scss to use SCSS nesting
* refactor integrations.scss to use SCSS nesting

rishig commented:
cool, thanks! The main comment I have is that I think the original styling looked better for /help. 
(In general I think the original /help formatting was well-balanced; just stuff like /why-zulip and /security were bad.)

alexandraciobica commented:
> cool, thanks! The main comment I have is that I think the original styling looked better for /help (and /api).
> In general I think the original /help formatting was well-balanced; just stuff like /why-zulip and /security were bad.

I also think that `/help` looks better with the original `font-size`.

There are two options:
* leave `.markdown` with `font size: 1.1rem` (as is this PR) and put a smaller font on `/help`
* keep as before `.markdown` with `font size: 1rem` and make it bigger on `why-zulip` and `integrations`.

timabbott commented:
I agree; I think doing the larger font-size off /help/ is fine for the near term; my guess is that we just need to do more careful styling work on details like line-height to make a larger font size look good on /help/, and we should defer that work until this is integrated.  Here's what I think we need to do in order to merge this PR:
* Fix the issue with numbered lists on /help/ being misformatted with this PR.
* Make the font size increase not apply to /help/; I think I prefer I prefer the implementation strategy of having `/help/` be the standard and other pages that need different sizing override that.
* (I think those are the only major regressions I'm aware of).

I expect we'll have a bunch of other near-term follow-ups, but it'll save a lot of time for everyone if we can integrate this and build on top of that, because re-reviewing long PRs where we're happy with almost everything in it gets very expensive (since I end up spending a lot of time re-reading the same commits).

alexandraciobica commented:
I made two changes since last push:
* fixed the `font-size` issue and this also fixed the `ordered lists` on `/help` (21fc7d9)
* the ordered lists on `integrations` were broken, I fixed them in the last commit (d0f6b25)
<img width="789" alt="Screen Shot 2019-07-24 at 14 06 26" src="https://user-images.githubusercontent.com/18381652/61789437-61d2cb80-ae1d-11e9-83ca-5671451d73d8.png">

<img width="561" alt="Screen Shot 2019-07-24 at 14 17 10" src="https://user-images.githubusercontent.com/18381652/61789611-ce4dca80-ae1d-11e9-8b17-b60701bd959e.png">

I will not make any more changes in this PR, I will include the rest of the changes in a follow up PR.

timabbott commented:
Awesome, merged, thanks @alexandraciobica!  I'm really excited about this direction.

I made one change, which was to make the last commit just change the 30px to 32px, rather than adding a page-specific override.  Though now that I think about it, maybe we should have changed it to something measured in `em` instead, since fundamentally this is an issue of scaling with font size.

A few ideas to add to your list from looking at the code here:
* `integrations.scss` seems like it has a decent amount of CSS duplicated with `.markdown` that we might be able to delete.  E.g. this:
```
.portico-landing.integrations ol li {                                                                
    list-style: none;                                                                                
}                                                                                                    
```
* That file is probably a good case for doing SCSS nesting.  Possibly this is something @vrongmeal can do since he's done a ton of that sort of migration so you can focus on more functional stuff, though we should be careful to coordinate when the SCSS migration happens to avoid merge conflicts.

And then there's also all the visual things we've been talking about.

timabbott commented:
@rishig we should probably also chat about next steps design-wise as a group.
